tests/functional/configs/test_versioned_model_constraint.py

```diff
@@ -0,0 +1,76 @@
+import pytest
+from dbt.tests.util import run_dbt, rm_file, write_file, get_manifest
+
+
+schema_yml = """
+models:
+  - name: foo
+    config:
+      materialized: table
+      contract:
+        enforced: true
+    constraints:
+      - type: primary_key
+        columns: [id, user_name]
+    columns:
+      - name: id
+        data_type: int
+        constraints:
+          - type: not_null
+      - name: user_name
+        data_type: text
+"""
+
+foo_sql = """
+select 1 as id, 'alice' as user_name
+"""
+
+versioned_schema_yml = """
+models:
+  - name: foo
+    latest_version: 1
+    config:
+      materialized: table
+      contract:
+        enforced: true
+    constraints:
+      - type: primary_key
+        columns: [id, user_name]
+    columns:
+      - name: id
+        data_type: int
+        constraints:
+          - type: not_null
+      - name: user_name
+        data_type: text
+    versions:
+      - v: 1
+"""
+
+
+class TestVersionedModelConstraints:
```

Comment:
Are we already testing versions column level constrains somewhere?

Reply:
I don't know. I don't see anything for it.